third_party/terraform/utils/common_diff_suppress.go.erb

```diff
@@ -18,6 +18,28 @@ func optionalPrefixSuppress(prefix string) schema.SchemaDiffSuppressFunc {
 	}
 }
 
+func ignoreMissingKeyInMap(key string) schema.SchemaDiffSuppressFunc {
+	return func(k, old, new string, d *schema.ResourceData) bool {
+		log.Printf("suppressing diff %q with old %q, new %q", k, old, new)
+		if strings.HasSuffix(k, ".%") {
+			oldNum, err := strconv.Atoi(old)
+			if err != nil {
+				log.Printf("[ERROR] could not parse %q as number, no longer attempting diff suppress", old)
+				return false
+			}
+			newNum, err := strconv.Atoi(new)
+			if err != nil {
+				log.Printf("[ERROR] could not parse %q as number, no longer attempting diff suppress", new)
+				return false
+			}
+			return oldNum+1 == newNum
```

Comment:
Maybe I'm missing some context on why this is needed? I looked at the original issue and the if statement below where we check that `x-goog-version` is defined should be sufficient. Isn't the issue that the customer doesn't define this in their config and the api hands it back? Maybe we can sync on this offline (well I guess online because of covid :P)

Reply:
Ah, yeah, let me explain in chat.